packages/medusa/src/api/routes/store/products/list-products.ts

```diff
@@ -199,6 +199,12 @@ export default async (req, res) => {
 
   // get only published products for store endpoint
   filterableFields["status"] = ["published"]
+  // store APIs only receive active and public categories to query from
+  filterableFields["categories"] = {
```

Comment:
suggestion: Should we only do that if categories is present in fileterableFields?

Reply:
oh yeah true, we alawys want to return product from active categories right? And if a product is not part of any categories what result do we get?

Reply:
If you're querying the API with a category, it will not be returned. But if you're querying without a category, it will be returned.